### ISSUE TITLE
Timetable is hard to navigate

### ISSUE THREAD

**Issue by jugmac00** [canonical/canonical-indico-ngtimetable#16]
On Ubuntu 20.04, with Chromium, Firefox and Chrome, the timetable is hard to navigate, as you both need to scroll horizontally and vertically.

![Screenshot from 2023-09-27 17-13-14](https://github.com/canonical/canonical-indico-ngtimetable/assets/9895620/b19c776b-b022-46d8-8cd7-bdda59d2c6e3)

On top of that, the horizontal scrollbar is only visible when you scroll to the bottom of the page.

![Screenshot from 2023-09-27 17-16-25](https://github.com/canonical/canonical-indico-ngtimetable/assets/9895620/e8b9b351-3ff0-491d-89ba-34cec1014758)

It is also not easy possible to choose a day/date of the conference.

Additionally, scrolling with my mouse wheel feels odd - when I scroll down it stops before I reach the bottom of the page. I think this has something to do with either iframes or scrollable containers. I also saw a `scroll-snap` directive, which may be related to the issue, though I am no front end dev. This behavior is very unfortunate, as at that time I did not see the horizontal scroll bar, and I assumed I have already reached the bottom of the page.

![Screenshot from 2023-09-27 17-17-52](https://github.com/canonical/canonical-indico-ngtimetable/assets/9895620/812f4eaa-bd51-4046-8308-9212139175eb)
In above screenshot you can see the position when scrolling stopped although I kept using my mouse wheel - please note that I do not see a horizontal scroll bar yet.

**Comment by kewisch** [comment]
Alright, I might have some fixes for this. If you paste this into a new stylesheet in Firefox's style editor, does it work better? The overlapping text is not solved, I'll think about that one separately.

```css
.ngtimetable, .ngtimetable > .schedule {
  flex: 1 0 0 !important;
}
.ngtimetable > .schedule {
  scroll-snap-type: none !important;;
  display: grid;
  grid-template-columns: 40px repeat(99, auto);
  grid-template-rows: 1fr;
}

.rooms {
  height: auto !important;
}

.footer {
  display: none;
}
```

Scroll snap appears to work much more smoothly on Firefox. It would just jump a little bit, not as harsh as Chrome. This seems much more useful on mobile, I can restrict it to smaller screens.

Let's create a separate issue for navigation between days. Now that we have the menu we can easily add the weekdays to the menu. Note adding #saturday to the end will give you an anchor to Saturday

**Comment by jugmac00** [comment]
@kewisch Thank you! This works much, much better!

And TIL about that you can create a new stylesheet for FireFox in the devtools. Thank you for that as well!

I just found the "change orientation" button (top left, below the heading, in the menu where you can also zoom), and when I actually change the orientation, scrolling feels super weird. This is probably the same issue with `scroll-snap-type`.

**Comment by kewisch** [comment]
For  vertical you'll need this:

```css
.ngtimetable, .ngtimetable > .schedule {
  flex: 1 0 0 !important;
}
.ngtimetable > .schedule {
  scroll-snap-type: none !important;;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows:  40px repeat(99, auto);
}

.footer {
  display: none;
}
```


Getting closer to a solution ready for a commit, but TIL subgrid was only just supported by Chrome since beginning of the month.   (╯°□°)╯︵ ┻━┻

**Comment by kewisch** [comment]
To the extent possible as a non-webdev, I've fixed this in b190fd1f9f6b0c7fcf717d8a69b22e9f53ac8033.
